src/sparsezoo/objects/model_helpers.py

```diff
@@ -0,0 +1,159 @@
+from sparsezoo.utils import download_get_request
+
+
+__all__ = [
+    "load_files_from_stub",
+    "load_files_from_directory",
+    "generate_model_name",
+    "ZOO_STUB_PREFIX",
+]
+
+_LOGGER = logging.getLogger(__name__)
+
+ZOO_STUB_PREFIX = "zoo:"
+
+
+def load_files_from_directory(directory_path: str) -> List[Dict[str, Any]]:
+    """
+    :param directory_path: a path to the directory,
+        that contains model files in the expected structure
+    :return list of file dictionaries
+    """
+    display_names = os.listdir(directory_path)
+    if not display_names:
+        raise ValueError(
+            "The directory path is empty. "
+            "Check whether the indicated directory exists."
+        )
+    files = [
+        dict(display_name=display_name, path=os.path.join(directory_path, display_name))
+        for display_name in display_names
+    ]
+    return files
+
+
+def load_files_from_stub(
+    stub: str,
+    valid_params: Optional[List[str]] = None,
+    force_token_refresh: bool = False,
+) -> Tuple[List[Dict[str, Any]], Dict[str, str]]:
+    """
+    :param stub: the SparseZoo stub path to the model (optionally
+        may include string arguments)
+    :param valid_params: list of expected parameter names to be encoded in the
+        stub. Will raise a warning if any unexpected param names are given. Leave
+        as None to not raise any warnings. Default is None
+    :param force_token_refresh: True to refresh the auth token, False otherwise
+    :return: The tuple of
+        - list of file dictionaries
+        - parsed param dictionary
+    """
+    if isinstance(stub, str):
+        stub, params = parse_zoo_stub(stub, valid_params=valid_params)
+    _LOGGER.debug(f"load_model_from_stub: loading model from {stub}")
+    response_json = download_get_request(
+        args=stub,
+        force_token_refresh=force_token_refresh,
+    )
+    # piece of code required for backwards compatibility
+    files = restructure_request_json(response_json["model"]["files"])
+    if params:
+        files = filter_files(files, params)
+    return files, params
+
+
+def filter_files(
+    files: List[Dict[str, Any]], params: Dict[str, str]
+) -> List[Dict[str, Any]]:
+    """
+    Use the `params` to extract only the relevant files from `files`
+
+    :param files: a list of file dictionaries
+    :param params: a dictionary with filtering parameters
+    :return a filtered `files` object
+    """
+    available_params = set(params.keys())
+    files_filtered = []
+    for file_dict in files:
+        if "recipe" in available_params and file_dict["file_type"] == "recipe":
+            value = params["recipe"]
+            if file_dict["display_name"] != "recipe_" + value + ".md":
+                continue
+        if "checkpoint" in available_params and file_dict["file_type"] == "training":
+            pass
+
+        if "deployment" in available_params and file_dict["file_type"] == "deployment":
+            pass
```

Comment:
In theory, we refrain from adding todos. There will be an explicit need to revisit those functionalities once the server migration is complete.

Reply:
Huh why refrain from adding todos? Is it captured somewhere else? Otherwise we have to keep this info in our heads and people who don't know will be confused when they come across these blocks.

Reply:
Hmm. Since I we are merging this PR into feature branch and not main, maybe adding a todo here to have a "mental marker" is actually reasonable. @bfineran ?

Reply:
@corey-nm the general rule is no todos in the production code.

Reply:
How about `NOTE: these will be filled out in future releases and are left here as a reminder`? 😁